### PR TITLE
Add pool function to sdk{PFT-553]

### PR DESCRIPTION
**Pull request by iamamitkumar0512**

### CODE REVIEW

File: src/subgraph/index.ts [hunk 1]
```diff
@@ -134,4 +143,59 @@ export class Subgraph {
...
+    return getChainVaultData(this.rpcConfig.chainId, subgraphEndpoint);
+  }
+
+  public async getTotalPoolValue() {
+    await this.init();
+    const subgraphEndpoint = this.getSubgraphEndpoint(this.rpcConfig.chainId);
+    const vaults = await getChainVaultData(this.rpcConfig.chainId, subgraphEndpoint);
+    const priceIds = vaults.map((v) => v.depositToken?.pyth?.id);
+    const res = await this.pyth.getLatestPricesFromPyth(priceIds as string[]);
+    const price = res.map((pythPrice) => {
```
Comment (wafflemakr) [inline src/subgraph/index.ts]: Can you name this variable like priceInfos or something more meaninful? as `price` is like for one id only, but this is an array.

File: src/subgraph/index.ts [hunk 2]
```diff
@@ -134,4 +143,59 @@ export class Subgraph {
     const subgraphEndpoint = this.getSubgraphEndpoint(this.rpcConfig.chainId);
     return getAllVaults(subgraphEndpoint);
   }
+
+  public async getChainVaultData(): Promise<Vault[]> {
+    const subgraphEndpoint = this.getSubgraphEndpoint(this.rpcConfig.chainId);
+    return getChainVaultData(this.rpcConfig.chainId, subgraphEndpoint);
+  }
+
+  public async getTotalPoolValue() {
```
Comment (wafflemakr) [inline src/subgraph/index.ts]: Please add comments to all functions and lets rename it to `getTotalPoolsValue`, as its for all pools, not just one

File: src/subgraph/index.ts [hunk 3]
```diff
@@ -134,4 +143,59 @@ export class Subgraph {
...
+        }
+      }
+      // Return null if no matching priceId found
+      return null;
+    }
+
+    const data = vaults.map((vault) => {
+      const normalizedPrice = getNormalizedPriceById(vault.depositToken?.pyth?.id as string, price);
+      const totatVaultValue =
+        (Number(vault.totalAssets) / 10 ** (vault.vaultDecimals || 0)) * Number(normalizedPrice || 0);
```
Comment (wafflemakr) [inline src/subgraph/index.ts]: More likely that the vaultDecimals is 18 than 0, so lets do 18 instead

File: src/subgraph/index.ts [hunk 4]
```diff
@@ -134,4 +143,59 @@ export class Subgraph {
...
+    const vaults = await getChainVaultData(this.rpcConfig.chainId, subgraphEndpoint);
+    const priceIds = vaults.map((v) => v.depositToken?.pyth?.id);
+    const res = await this.pyth.getLatestPricesFromPyth(priceIds as string[]);
+    const price = res.map((pythPrice) => {
+      const normalizedPrice = this.pyth.normalizePythPriceForParifi(
+        Number(pythPrice.price.price),
+        pythPrice.price.expo,
+      );
+      const colletralPrice = new Decimal(normalizedPrice).div(10 ** PRICE_FEED_DECIMALS);
+      const returnObj = { priceId: `0x${pythPrice.id}`, normalizedPrice: colletralPrice };
```
Comment (wafflemakr) [inline src/subgraph/index.ts]: typo

File: src/subgraph/index.ts [hunk 5]
```diff
@@ -134,4 +143,59 @@ export class Subgraph {
...
+          // Return the normalizedPrice if matched
+          return price.normalizedPrice;
+        }
+      }
+      // Return null if no matching priceId found
+      return null;
+    }
+
+    const data = vaults.map((vault) => {
+      const normalizedPrice = getNormalizedPriceById(vault.depositToken?.pyth?.id as string, price);
```
Comment (wafflemakr) [inline src/subgraph/index.ts]: if this returns null, then lets set it to 0 instead or handle that scenario better

**Review by wafflemakr — CHANGES_REQUESTED** [review]

**Comment by iamamitkumar0512** [comment]
image
image

File: src/subgraph/vaults/index.ts [hunk 6]
```diff
@@ -17,3 +33,111 @@ export const getAllVaults = async (subgraphEndpoint: string): Promise<Vault[]> =
...
+      priceInfos,
+    );
+    const vaultPerShare = vault.vault.assetsPerShare;
+    const userShare = vault.sharesBalance;
+    const myBalance =
+      (Number(userShare || 0) * Number(vaultPerShare || 0)) /
+      Number(10 ** (vault.vault?.vaultDecimals || 18)) /
+      Number(10 ** parseInt(vault.vault.depositToken?.decimals || ''));
+
+    const totatVaultValue = myBalance * Number(normalizedPrice);
```
Comment (sudeepb02) [inline src/subgraph/vaults/index.ts]: Fix typo for `totatVaultValue` - total

File: src/subgraph/index.ts (hunk 5 above)
Comment (iamamitkumar0512) [reply src/subgraph/index.ts]: done

File: src/subgraph/index.ts (hunk 4 above)
Comment (iamamitkumar0512) [reply src/subgraph/index.ts]: fixed

File: src/subgraph/index.ts (hunk 3 above)
Comment (iamamitkumar0512) [reply src/subgraph/index.ts]: done

File: src/subgraph/index.ts (hunk 2 above)
Comment (iamamitkumar0512) [reply src/subgraph/index.ts]: done

File: src/subgraph/index.ts (hunk 1 above)
Comment (iamamitkumar0512) [reply src/subgraph/index.ts]: done

**Review by sudeepb02 — CHANGES_REQUESTED** [review]
@iamamitkumar0512 Please fix the typo

File: src/subgraph/vaults/index.ts (hunk 6 above)
Comment (iamamitkumar0512) [reply src/subgraph/vaults/index.ts]: Done

**Review by sudeepb02 — APPROVED** [review]

File: src/subgraph/vaults/index.ts [hunk 7]
```diff
@@ -17,3 +29,111 @@ export const getAllVaults = async (subgraphEndpoint: string): Promise<Vault[]> =
...
+      priceId: vault.vault.depositToken?.pyth?.id,
+      totalVaultValue: totalVaultValue,
+    };
+    return returnObj;
+  });
+  const myTotalPoolValue = data.reduce((a, b) => a + b.totalVaultValue, 0);
+  return { data, myTotalPoolValue };
+};
+
+export const getTotalPoolsValue = async (
```
Comment (wafflemakr) [inline src/subgraph/vaults/index.ts:109]: Cant we merge these two functions, and have an optional param `user` instead? this way we avoid having to call the subgraph twice..

Comment (iamamitkumar0512) [reply src/subgraph/vaults/index.ts:109]: I am using this query from interface

**Review by wafflemakr — APPROVED** [review]

**Pull request merged**